src/main/java/gtPlusPlus/xmod/gregtech/common/helpers/ChargingHelper.java

```diff
@@ -33,125 +35,91 @@
+
+                    // Utils.LOG_WARNING("Found Player.");
+
+                    if (Utils.isServer()) {
+                        // Utils.LOG_WARNING("Found Server-Side.");
+
+                        mTickTimer++;
+                        if (mTickTimer % mTickMultiplier == 0) {
+
+                            long mVoltage;
+                            long mEuStored;
+
+                            if (!mChargerMap.isEmpty() && mValidPlayers.containsKey(mPlayerMan.getDisplayName())) {
+                                InventoryPlayer mPlayerInventory = mPlayerMan.inventory;
+                                ItemStack[] mArmourContents = mPlayerInventory.armorInventory.clone();
+                                ItemStack[] mInventoryContents = mPlayerInventory.mainInventory.clone();
+                                ItemStack[] baubleSlots = null;
+                                if (Baubles.isModLoaded()) {
+                                    IInventory baubleInv = BaublesApi.getBaubles(mPlayerMan);
+                                    if (baubleInv != null) {
+                                        baubleSlots = new ItemStack[baubleInv.getSizeInventory()];
+                                        for (int i = 0; i < baubleInv.getSizeInventory(); i++) {
+                                            baubleSlots[i] = baubleInv.getStackInSlot(i);
+                                        }
+                                    }
+                                }
+
+                                for (GregtechMetaWirelessCharger mEntityTemp : mChargerMap.values()) {
+                                    if (mEntityTemp != null) {
+                                        if (mEntityTemp.getBaseMetaTileEntity() == null
+                                                || !mEntityTemp.getBaseMetaTileEntity().isAllowedToWork())
+                                            continue;
+                                        if (mPlayerMan.getEntityWorld().provider.dimensionId
+                                                == mEntityTemp.getDimensionID()) {
+                                            mVoltage = mEntityTemp.maxEUInput();
+                                            mEuStored = mEntityTemp.getEUVar();
+                                            if (mVoltage > 0 && mEuStored >= mVoltage) {
+
+                                                Map<String, UUID> LR = mEntityTemp.getLongRangeMap();
+                                                Map<String, UUID> LO = mEntityTemp.getLocalMap();
+
+                                                long mStartingEu = mEntityTemp.getEUVar();
+                                                long mCurrentEu = mEntityTemp.getEUVar();
+                                                if (canCharge(mEntityTemp, mPlayerMan, LR, LO)) {
+                                                    mCurrentEu -= chargeItems(mEntityTemp, mArmourContents, mPlayerMan);
```

Comment:
`chargeItems` will return stored energy in the tile, so substracting it doesn't make sense. Maybe use `if (mStartingEu - mEntityTemp.getEUVar() <= 0)`